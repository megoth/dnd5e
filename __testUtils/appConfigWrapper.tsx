import React, { ReactNode } from "react";
import AppConfigProvider from "../src/contexts/appConfig";
import mockAppConfig from "./mockAppConfig";

interface Props {
  children: ReactNode;
}

export default function AppConfigWrapper({ children }: Props) {
  const { errorsUrl, translationsUrl } = mockAppConfig();
  return (
    <AppConfigProvider errorsUrl={errorsUrl} translationsUrl={translationsUrl}>
      {children}
    </AppConfigProvider>
  );
}
