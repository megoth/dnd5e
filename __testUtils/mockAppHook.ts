import mockApp from "./mockApp";

export default function mockAppHook(mock: jest.Mock, app = mockApp()) {
  mock.mockReturnValue(app);
}
