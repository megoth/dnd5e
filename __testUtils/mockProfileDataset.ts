import {
  addStringNoLocale,
  addUrl,
  mockSolidDatasetFrom,
  mockThingFrom,
  setThing,
} from "@inrupt/solid-client";
import { vcard, foaf, rdf } from "rdf-namespaces";
import { chain } from "../src/utils";

export const aliceWebIdDatasetUrl = "http://example.com/alice";
export const aliceWebIdUrl = `${aliceWebIdDatasetUrl}#me`;
export const aliceName = "Alice";
export const aliceNick = "A";
export const alicePhoto = "http://example.com/alice.jpg";

export function mockProfileDataset(webId = aliceWebIdUrl) {
  const profile = chain(
    mockThingFrom(webId),
    (t) => addStringNoLocale(t, vcard.fn, aliceName),
    (t) => addStringNoLocale(t, vcard.nickname, aliceNick),
    (t) => addUrl(t, vcard.hasPhoto, alicePhoto),
    (t) => addUrl(t, rdf.type, foaf.Person),
    (t) => addUrl(t, foaf.openid, aliceWebIdUrl)
  );
  return chain(mockSolidDatasetFrom(aliceWebIdDatasetUrl), (d) =>
    setThing(d, profile)
  );
}
