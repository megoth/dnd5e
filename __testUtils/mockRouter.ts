import { NextRouter } from "next/router";

export default function mockRouter(
  overrides: Partial<NextRouter> = {}
): NextRouter {
  return {
    asPath: "/",
    query: {},
    ...overrides,
  } as NextRouter;
}
