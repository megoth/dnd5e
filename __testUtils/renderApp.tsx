import React from "react";
import { render, RenderOptions, RenderResult } from "@testing-library/react";
import { LocalizationProvider, ReactLocalization } from "@fluent/react";
import { getFluentBundles } from "../src/models/language";
import { AppModel } from "../src/models/app";

export default function renderApp(
  app: AppModel,
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult {
  function rerender() {
    return render(
      <LocalizationProvider l10n={new ReactLocalization(getFluentBundles(app))}>
        {ui}
      </LocalizationProvider>,
      options
    );
  }
  const renderResult = rerender();
  return { ...renderResult, rerender };
}
