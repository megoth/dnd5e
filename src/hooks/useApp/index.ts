import { useContext, useEffect } from "react";
import { AppContext } from "../../contexts/app";
import { appIsLoadingLocalizations } from "../../models/app";

export default function useApp(bundles: string[] = []) {
  const { app, setBundles } = useContext(AppContext);
  useEffect(() => {
    if (!app || !appIsLoadingLocalizations(app, bundles)) {
      return;
    }
    setBundles(bundles);
  }, [app, bundles, setBundles]);
  return app;
}
