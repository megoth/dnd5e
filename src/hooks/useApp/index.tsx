import { useContext } from "react";
import AppContext from "./context";

export default function useApp() {
  return useContext(AppContext);
}
