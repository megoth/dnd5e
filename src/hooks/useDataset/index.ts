import useSWR from "swr";
import { getSolidDataset } from "@inrupt/solid-client";
import { useSession } from "@inrupt/solid-ui-react";

export default function useDataset(url) {
  const { fetch } = useSession();
  return useSWR([url, "dataset"], () => getSolidDataset(url, { fetch }));
}
