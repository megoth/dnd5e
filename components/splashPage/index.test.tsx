import React from "react";
import * as routerFns from "next/router";
import useApp from "../../src/hooks/useApp";
import mockAppHook from "../../__testUtils/mockAppHook";
import useDataset from "../../src/hooks/useDataset";
import { mockProfileDataset } from "../../__testUtils/mockProfileDataset";
import SplashPage from "./index";
import mockRouter from "../../__testUtils/mockRouter";
import renderApp from "../../__testUtils/renderApp";

jest.mock("../../src/hooks/useApp");
const mockedAppHook = useApp as jest.Mock;

jest.mock("../../src/hooks/useDataset");
const mockedUseDataset = useDataset as jest.Mock;

describe("SplashPage", () => {
  beforeEach(() => {
    jest
      .spyOn(routerFns, "useRouter")
      .mockReturnValue(mockRouter({ asPath: "/" }));
  });

  it("renders", () => {
    const app = mockAppHook(mockedAppHook);
    mockedUseDataset.mockReturnValue(mockProfileDataset());
    const { asFragment } = renderApp(app, <SplashPage />);
    expect(asFragment()).toMatchSnapshot();
  });
});
