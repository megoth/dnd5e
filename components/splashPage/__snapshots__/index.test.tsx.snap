// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SplashPage renders 1`] = `
<DocumentFragment>
  <div
    class="flex-1 flex flex-col relative min-h-full"
  >
    <main>
      <section
        class="hero"
      >
        <header
          class="pb-1 bg-white dark:bg-gray-800"
        >
          <div
            class="main-container main-container--content flex"
          >
            <a
              class="focus:outline-none focus:ring-2 focus:ring-red-600"
              href="/"
            >
              <picture
                class="block"
              >
                <source
                  media="(min-width: 640px)"
                  srcset="/icons/app/icons8-dungeons-and-dragons-128.png"
                />
                <source
                  media="(min-width: 480px)"
                  srcset="/icons/app/icons8-dungeons-and-dragons-96.png"
                />
                <img
                  alt="[Translation for https://example.com/global/translations.ttl#appLogoDescription does not exist]"
                  src="/icons/app/icons8-dungeons-and-dragons-64.png"
                />
              </picture>
            </a>
            <div
              class="flex-1"
            >
              <div
                class="font-serif self-center"
              >
                <a
                  class="focus:outline-none focus:ring-2 focus:ring-red-600"
                  href="/"
                >
                  [Translation for https://example.com/global/translations.ttl#appName does not exist]
                </a>
              </div>
              <nav
                class="pages-nav pages-nav--main-menu"
              >
                <ul
                  class="pages-nav__list pages-nav__list--main-menu"
                >
                  <li
                    class="flex"
                    data-testid="main-nav-list-item"
                  >
                    <a
                      class="pages-nav__link pages-nav__link--main-menu"
                      href="/characters"
                    >
                      [Translation for https://example.com/global/translations.ttl#charactersPageTitle does not exist]
                    </a>
                  </li>
                  <li
                    class="flex"
                    data-testid="main-nav-list-item"
                  >
                    <a
                      class="pages-nav__link pages-nav__link--main-menu"
                      href="/rules"
                    >
                      [Translation for https://example.com/global/translations.ttl#rulesPageTitle does not exist]
                    </a>
                  </li>
                </ul>
              </nav>
            </div>
          </div>
        </header>
        <div
          class="flex-1 flex flex-col justify-center items-center leading-normal px-4 bg-gradient-to-t from-white dark:from-gray-800 to-white dark:to-gray-800 via-transparent dark:via-transparent relative text-center"
        >
          <div
            class="p-4 py-2 bg-white dark:bg-gray-800 bg-opacity-75 dark:bg-opacity-90 rounded max-w-lg"
          >
            <h1
              class="text-3xl xs:text-4xl sm:text-6xl my-2 font-serif font-semibold"
            >
              [Translation for https://example.com/global/translations.ttl#appName does not exist]
            </h1>
            <p
              class="my-1 max-w-xl"
            >
              [Translation for https://example.com/global/translations.ttl#appPitch does not exist]
            </p>
            <div
              class="bg-yellow-200 text-yellow-900 border-yellow-400 border px-2 py-1 rounded-sm mb-2"
            >
              [Translation for https://example.com/global/translations.ttl#workInProgress does not exist]
            </div>
            <a
              class="link block my-2 max-w-prose text-center"
              href="/about"
            >
              [Translation for https://example.com/global/translations.ttl#learnMore does not exist]
            </a>
          </div>
          <p
            class="text-xs text-gray-600 absolute bottom-1 right-2"
          >
            Photo by 
            <a
              class="link link--subtle"
              href="https://unsplash.com/@steve_j?utm_source=unsplash&utm_medium=referral&utm_content=creditCopyText"
            >
              Steve Johnson
            </a>
             on 
            <a
              class="link link--subtle"
              href="https://unsplash.com/s/photos/pen-and-paper?utm_source=unsplash&utm_medium=referral&utm_content=creditCopyText"
            >
              Unsplash
            </a>
          </p>
        </div>
      </section>
      <section
        class="text-center my-16 px-4 sm:px-0 flex flex-col sm:flex-row max-w-prose mx-auto"
      >
        <div
          class="content leading-normal max-w-prose mx-auto flex-1 flex flex-col justify-center"
          style="hyphens: manual;"
        >
          <img
            alt="[Translation for https://example.com/global/translations.ttl#solidLogo does not exist]"
            class="mx-auto"
            src="/logos/solid-emblem.svg"
            style="max-height: 175px; max-width: 200px;"
          />
          <h2
            class="px-4"
          >
            [Translation for https://example.com/global/translations.ttl#logInPitch does not exist]
          </h2>
        </div>
        <div
          class="flex-1 flex flex-col justify-center"
        >
          <div
            class="content leading-normal max-w-prose mx-auto mb-4"
            style="hyphens: auto;"
          >
            <p>
              [Translation for https://example.com/global/translations.ttl#authenticationGuidance does not exist]
            </p>
          </div>
          <ul
            class="flex flex-col space-y-2 xs:max-w-xs md:max-w-full mx-auto"
          >
            <li
              class="flex-1"
            >
              <button
                class="button button--solid block w-full"
                data-testid="login-button"
                type="button"
              >
                Inrupt Pod Spaces
              </button>
            </li>
            <li
              class="flex-1"
            >
              <button
                class="button button--solid block w-full"
                data-testid="login-button"
                type="button"
              >
                solidcommunity.net
              </button>
            </li>
            <li
              class="flex-1"
            >
              <button
                class="button button--solid block w-full"
                data-testid="login-button"
                type="button"
              >
                inrupt.net
              </button>
            </li>
            <li
              class="flex-1"
            >
              <a
                class="button button--solid block w-full"
                href="/login?idp"
              >
                [Translation for https://example.com/global/translations.ttl#provideIdP does not exist]
              </a>
            </li>
          </ul>
          <div
            class="content leading-normal max-w-prose mx-auto mt-4"
            style="hyphens: auto;"
          >
            <a
              href="/signup"
            >
              [Translation for https://example.com/global/translations.ttl#signupGuidance does not exist]
            </a>
          </div>
        </div>
      </section>
      <footer
        class="main-container main-container--content mt-8 text-center sm:text-left xs:flex xs:space-x-4 items-center justify-center sm:justify-start"
      >
        <div
          class="md:flex-grow pb-2 flex space-x-1 justify-center sm:justify-start"
        >
          <a
            class="link"
            href="/about"
          >
            [Translation for https://example.com/global/translations.ttl#aboutTitle does not exist]
          </a>
          <a
            class="link"
            href="/faq"
          >
            [Translation for https://example.com/global/translations.ttl#faqShort does not exist]
          </a>
        </div>
        <div
          class="pb-2 flex xs:space-x-2 space-y-2 xs:space-y-0 flex-col text-center xs:flex-row xs:justify-center"
        >
          <button
            class="button"
            data-testid="dark-mode-selector-button"
            type="button"
          >
            <span>
              ðŸŒ™Â [Translation for https://example.com/global/translations.ttl#darkModeTurnOn does not exist]
            </span>
          </button>
        </div>
      </footer>
    </main>
  </div>
</DocumentFragment>
`;
